web/src/app/pages/create-survey/create-survey.component.ts

```diff
@@ -51,39 +50,105 @@ export class CreateSurveyComponent implements OnInit {
       }
       this.currentSurveyId = surveyId;
     });
-    this.surveyService
-      .getActiveSurvey$()
-      .subscribe(survey => this.loadExistingSurvey(survey));
+    this.surveyService.getActiveSurvey$().subscribe(survey => {
+      this.currentSurvey = survey;
+      this.setupPhase = this.decideSetupPhase(survey);
+    });
+  }
+
+  private decideSetupPhase(survey: Survey): SetupPhase {
+    if (survey.title && survey.title.trim().length > 0) {
+      return SetupPhase.NAME_JOB;
+    }
+    return SetupPhase.NAME_SURVEY;
+  }
+
+  job(): Job | undefined {
+    if (this.currentSurvey?.jobs.size ?? 0 > 0) {
+      return this.currentSurvey?.jobs.values().next().value;
+    }
+    return undefined;
+  }
+
+  isNameSurvey(): boolean {
+    return this.setupPhase === SetupPhase.NAME_SURVEY;
   }
 
-  loadExistingSurvey(survey: Survey): void {
-    this.formGroup.controls[this.titleControlKey].setValue(survey.title);
-    this.formGroup.controls[this.descriptionControlKey].setValue(
-      survey.description
-    );
+  isNameJob(): boolean {
+    return this.setupPhase === SetupPhase.NAME_JOB;
   }
 
   back(): void {
-    this.navigationService.navigateToSurveyList();
+    switch (this.setupPhase) {
+      case SetupPhase.NAME_SURVEY:
+        this.navigationService.navigateToSurveyList();
+        break;
+      case SetupPhase.NAME_JOB:
+        this.setupPhase = SetupPhase.NAME_SURVEY;
+        break;
+      default:
+        break;
+    }
   }
 
   async continue(): Promise<void> {
-    const title = this.formGroup.controls[this.titleControlKey].value;
-    const description =
-      this.formGroup.controls[this.descriptionControlKey].value;
+    switch (this.setupPhase) {
+      case SetupPhase.NAME_SURVEY:
+        this.saveSurveyTitleAndDescription();
+        break;
+      case SetupPhase.NAME_JOB:
+        this.saveJobName();
+        break;
+      default:
+        break;
+    }
+  }
+
+  @ViewChild('nameSurvey')
+  nameSurvey: NameSurveyComponent | undefined;
+
+  private async saveSurveyTitleAndDescription(): Promise<void> {
+    const titleAndDescription = this.nameSurvey!.toTitleAndDescription();
     if (this.currentSurveyId) {
       await this.surveyService.updateTitleAndDescription(
         this.currentSurveyId,
-        title,
-        description
+        titleAndDescription[0],
+        titleAndDescription[1]
       );
-      this.navigationService.navigateToCreateJob(this.currentSurveyId);
     } else {
       const createdSurveyId = await this.surveyService.createSurvey(
-        title,
-        description
+        titleAndDescription[0],
+        titleAndDescription[1]
+      );
+      this.navigationService.navigateToCreateSurvey(createdSurveyId);
+    }
+  }
+
+  @ViewChild('nameJob')
+  nameJob: NameJobComponent | undefined;
```

Comment:
n00b question: can we user `?` instead of `| undefined` here?

Reply:
Oh, I tried `nameJob: NameJobComponent?;` and it did not compile, and it should be `nameJob?: NameJobComponent;`. Done.